app/gui2/src/components/GraphNode.vue

```diff
@@ -262,18 +297,55 @@ function handleClick(e: PointerEvent) {
   }
 }
 
-const isCircularMenuVisible = ref(false)
-const isAutoEvaluationDisabled = ref(false)
-const isDocsVisible = ref(false)
-const isVisualizationVisible = ref(false)
+function updatePreprocessor(module: string, method: string, ...args: string[]) {
+  visualizationConfiguration.value = {
+    executionContextId: props.executionContextId,
+    visualizationModule: props.mainModule,
+    expression: { module, definedOnType: module, name: method },
+    ...(args.length !== 0 ? { positionalArgumentsExpressions: args } : {}),
+  }
+}
 
-const visualizationType = ref('Scatterplot')
-const visualization = shallowRef<Visualization>()
+watch(isVisualizationVisible, async (visible) => {
+  if (!visible) {
+    return
+  }
+  // TODO: should the old vis be detached?
+  await props.languageServer.attachVisualization(
+    visualizationId.value,
+    props.node.rootSpan.id,
+    visualizationConfiguration.value ?? {
+      executionContextId: props.executionContextId,
+      ...DEFAULT_VISUALIZATION_CONFIGURATION,
+    },
+  )
+})
```

Comment:
Detaching is important. Otherwise we will be constantly adding work on the language server side. You could use `onCleanup` of `watchEffect` here to make it a bit simpler. Just make sure to call it before first `await`.

Though it probably should be mostly handled elsewhere anyway (see next comments).